staging/src/k8s.io/client-go/transport/round_trippers.go

```diff
@@ -113,6 +113,45 @@ func (rt *authProxyRoundTripper) RoundTrip(req *http.Request) (*http.Response, e
 	return rt.rt.RoundTrip(req)
 }
 
+// legalHeaderRune reports whether r is a valid rune in a header
+// field name. RFC 7230 says:
+//   header-field   = field-name ":" OWS field-value OWS
+//   field-name     = token
+//   tchar = "!" / "#" / "$" / "%" / "&" / "'" / "*" / "+" / "-" / "." /
+//           "^" / "_" / "`" / "|" / "~" / DIGIT / ALPHA
+//   token = 1*tchar
+func legalHeaderRune(r rune) bool {
+	return int(r) < len(legalHeaderRunes) && legalHeaderRunes[r]
+}
+
+func percentEncodeByte(b byte) string {
+	return fmt.Sprintf("%%%x", b)
+}
+
+func percentEncodeRune(r rune) string {
```

Comment:
A rune is always a byte so you can make this just be
```go
return fmt.Sprintf("%%%x", byte(r))
```

Reply:
Runes are [int32s](https://golang.org/doc/go1#rune), but if we are OK with restricting extra key strings to the asciii subset of utf-8, that'd be OK. I deliberately encoded entire runes here, rather than bytes, to avoid extra-garbled wire-format strings.

Reply:
Go strings *are* utf-8, but utf-8 runes can be 1-4 bytes long ;)